### PR TITLE
Remix plugin api

### PR DESCRIPTION
**Pull request by GrandSchtroumpf**
Use `ApiFactory` to manage plugins.

### CODE REVIEW

File: src/app/tabs/compile-tab.js [hunk 1]
```diff
@@ -131,22 +138,8 @@ class CompileTab {
...
-      description: 'compile solidity contracts',
-      kind: 'compile'
-    }
-  }
-
   getCompilationResult () {
```
Comment (yann300) [inline src/app/tabs/compile-tab.js]: should this be tagged as `async` ?

File: src/app/tabs/compile-tab.js [hunk 2]
```diff
@@ -131,22 +138,8 @@ class CompileTab {
...
-    return new Promise((resolve, reject) => {
-      resolve(this.compileTabLogic.compiler.lastCompilationResult)
-    })
+    return this.compileTabLogic.compiler.lastCompilationResult
```
Comment (yann300) [inline src/app/tabs/compile-tab.js]: should this be tagged as async ?

**Comment by yann300** [comment]
for the browser test failure, please check if the following https://github.com/ethereum/remix-ide/pull/1763/files#diff-bd9c9dcd314f2d7df52935b3a6a4d504R505
is ok. This is where we start debugging when user click on `debug`. seems that browser test fails here...

File: src/app/tabs/compile-tab.js (hunk 2 above)
Comment (GrandSchtroumpf) [reply src/app/tabs/compile-tab.js]: no need. With the new way of handling request (the queue) the `async` is not needed for synchronous methods.

**Pull request merged**